packages/documentsearch-extension/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { ISearchProviderRegistry } from '@jupyterlab/documentsearch';
+
+import { IRetroShell } from '@retrolab/application';
+import { Widget } from '@lumino/widgets';
+
+const SEARCHABLE_CLASS = 'jp-mod-searchable';
+
+const retroShellWidgetListener: JupyterFrontEndPlugin<void> = {
+  id: '@retrolab/documentsearch-extension:retroShellWidgetListener',
+  requires: [IRetroShell, ISearchProviderRegistry],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    retroShell: IRetroShell,
+    registry: ISearchProviderRegistry
+  ) => {
```

Comment:
Reimplementation of https://github.com/jupyterlab/jupyterlab/blob/a8dff39a1a2be850a80861e05145e763079b8794/packages/documentsearch-extension/src/index.ts#L38 for retrolab